scipy/integrate/_quadpack_py.py

```diff
@@ -1168,7 +1181,19 @@ def nquad(func, ranges, args=None, opts=None, full_output=False):
         opts = [_OptFunc(opts)] * depth
     else:
         opts = [opt if callable(opt) else _OptFunc(opt) for opt in opts]
-    return _NQuad(func, ranges, opts, full_output).integrate(*args)
+    res = _NQuad(func, ranges, opts, True).integrate(*args)
+
+    if full_output is not None:
+        msg = ("Use of parameter `full_output` is deprecated. Please leave "
+               "`full_output` unspecified; `neval` can now be accessed as an "
+               "attribute of the object returned by `scipy.integrate.nquad`.")
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+
+    if full_output:
+        return res
```

Comment:
Is this not slightly contradictory to documentation in that the user is returned a tuple when they were expecting a result object?

Reply:
I don't see a need to advertise all the deprecated possibilities of `full_output` in the documentation. 
Users who write new code following the recommendation of not specifying `full_output` will get the result object described, and existing code will continue to work.